build_tools/python/e2e_model_tests/cmake_generator.py

```diff
@@ -26,6 +26,10 @@ def generate_rules() -> List[str]:
     runner_args = run_module_utils.build_run_flags_for_model(
         model=model,
         model_input_data=test_config.input_data) + test_config.extra_test_flags
+    # TODO(#11136): We should pick up the execution args from
+    # `build_run_flags_for_execution_config`. But currently the DRIVER needs to
+    # be populated separately in the CMake rule.
+    runner_args.append("--device_allocator=caching")
```

Comment:
Wat. This seems not great and I don't understand why we need to hardcode this

Reply:
It's because the CMake rule we use `iree_benchmark_suite_module_test` -> `iree_run_module_test` -> `iree_native_test` use a separate `DRIVER` arg for populating flags and test label.

I rewrote to give `build_run_flags_for_execution_config` an option to generate flags without driver.